src/Carbon/Carbon.php

```diff
@@ -3326,16 +3326,21 @@ public function isBirthday(Carbon $dt = null)
      */
     public function modify($modify)
     {
-        if ($this->local) {
-            return parent::modify($modify);
-        }
+        // If we changing time in some timezone with daylight correction, there can be the case,
+        // that this correction influence to time in nonwanted way.
+        // So we have to make this modification in UTC timezone.
+        if (!preg_match('/(sec|second|min|minute|hour)s?/i', $modify)) {
```

Comment:
Technically, this regex is identical to `/sec|min|hour/i`

Reply:
Also, what is the situation concerning the other modifiers ("day", "week", etc.)?

Reply:
Technically - yes, but this is more explicit and obvious for human reading. I will move the long regexp into comments and replace it by short one in the code.

Others is OK. The timezone not influence to modifications with unit bigger than hour.

Reply:
I had to dig a bit to find this documentation page: [Relative Formats](http://php.net/manual/en/datetime.formats.relative.php)

Reply:
I realised that `sec` matches to "ordinal" `second` (like `first`, `second`, `third`). Hm. I am not sure about how to solve this. I have to think about that.

Reply:
I think i will try something like this: `/[+-]?[0-9]+ (sec|min|hour)\w*( ago)?/i`. What do you think?

Reply:
@vlakoff, Look at https://github.com/briannesbitt/Carbon/issues/883#issuecomment-339064288 plz.